Articles/FileSystemProviders/extending.md

```diff
@@ -0,0 +1,213 @@
+                var rootPath = hostingEnvironment.MapPathWebRoot(folderLocation);
+                var rootUrl = hostingEnvironment.ToAbsolute(folderLocation);
+
+                return new PhysicalFileSystem(
+                    factory.GetRequiredService<IIOHelper>(),
+                    hostingEnvironment,
+                    factory.GetRequiredService<ILogger<PhysicalFileSystem>>(),
+                    rootPath,
+                    rootUrl);
+            });
+        }
+    }
+```
+
+When creating a PhyscialFileSystem it takes some dependencies like IIOHelper, but the last two parameters are what we're interested in. The rootPath is where your media will be stored on the disk, since netcore by default store files in the wwwroot, we must put our desired folder somewhere within wwwroot, to ensure that we use `hostingEnvironment.MapPathWebRoot(~/CustomMediaFolder)`, the ~ will be mapped to your wwwroot folder, so the final rootPath will be your/project/path/wwwroot/CustomMediaFolder, the ~ is therefore important!
+
+The other part of the puzzle is the rootUrl, which is the base URL your media files will be served from, so in this case, your image URL could look something like mysite.com/CustomMediaFolder/MyAwesomePicture.png, again the ~ is important. Another thing worth mentioning with the rootUrl is that it must be the same as the folder location, otherwise, you will get 404's for your images.
+
+This is all great if you want to change the location within the wwwroot folder of your project, but what if you want to store the media files outside wwwroot? This is fortunately possible but requires an extra step.
+
+As mentioned netcore by default store static files, such as media and css, in the wwwroot folder, but we can register an additional location in the configure section of our startup
+
+In the `configure` method in startup.cs register a new static file location like so:
+
+```c#
+public void Configure(IApplicationBuilder app)
+{
+    ...
+
+    app.UseStaticFiles(new StaticFileOptions
+    {
+        FileProvider = new PhysicalFileProvider(Path.Combine("D:", "storage", "umbracoMedia")),
+        RequestPath = "/CustomPath"
+    });
+}
+```
+The PhysicalFileProvider takes a single parameter, the **RootPath** - the rooted, filesystem path, using directory separator chars, NOT ending with a separator // eg "c:" or "c:\path\to\site" or "\\server\path", the safest way to achieve this is using `Path.Combine`.
+
+
+You also have to specify the  **RequestPath** - the relative url, where the media will be served, using url separator chars, NOT ending with a separator / eg "" or "/Views" or "/Media".
+
+Now you can use your newly registered static file location as if it was wwwroot. Notice how you no longer need to use `hostingEnvironment.MapPathWebRoot(folderLocation)`, since you're no longer trying to map the location to somewhere within wwwroot, but instead use your newly registered static file location.
```

Comment:
```suggestion
Now you can use your newly registered static file location as if it was `wwwroot`. Notice how you no longer need to use `hostingEnvironment.MapPathWebRoot(folderLocation)`, since you're no longer trying to map the location to somewhere within `wwwroot`, but instead use your newly registered static file location.
```